pkg/utils/managedresources/builder/managedresources.go

```diff
@@ -31,20 +31,28 @@ import (
 
 // ManagedResource is a structure managing a ManagedResource.
 type ManagedResource struct {
-	client   client.Client
-	resource *resourcesv1alpha1.ManagedResource
+	client         client.Client
+	createOnUpdate bool
```

Comment:
> Just an initial question from my side: isn't it more clearer to use `createOnReconcile` instead of `createOnUpdate` because the variable is being used for creation in the context of the `Reconcile` function in the `builder` package?

Actually, I think something like `createIfNotExists` or `updateOnly` is even more clear. WDYT?

Reply:
The `createOnUpdate` term was derived from the `AllowCreateOnUpdate` functions ([example](https://github.com/gardener/gardener/blob/3e5ef9a4880f86a5e05b44bc24cecd3416451685/pkg/registry/core/shootstate/strategy.go#L79)), but I really don't have a strong preference 🙂